src/main/java/com/avispl/symphony/dal/communicator/aver/ptz/AverPTZCommunicator.java

```diff
@@ -0,0 +1,1885 @@
+		} else if (powerStatus.compareTo(PowerStatus.ON) == 0) {
+			advancedControllableProperties.add(createSwitch(Command.POWER.getName(), 1, PowerStatus.OFF.getName(), PowerStatus.ON.getName()));
+
+			// Zoom control
+			populateZoomControl(stats, advancedControllableProperties);
+
+			// Focus control
+			populateFocusControl(stats, advancedControllableProperties);
+
+			// AE control
+			populateAEControl(stats, advancedControllableProperties);
+
+			// WB control
+			populateWBControl(stats, advancedControllableProperties);
+
+			// Pan tilt control
+			populatePanTiltControl(stats, advancedControllableProperties);
+
+			// Preset control
+			populatePresetControl(stats, advancedControllableProperties);
+		}
+	}
+
+	//region Control device
+
+	/**
+	 * This method is used to control image process
+	 */
+	private void imageProcessControl(String value, String[] splitProperty) {
+		// RGain
+		if (splitProperty[1].equals(Command.RGAIN.getName() + RGainControl.UP.getName())) {
+			rGainControl(RGainControl.UP);
+			return;
+		} else if (splitProperty[1].equals(Command.RGAIN.getName() + RGainControl.DOWN.getName())) {
+			rGainControl(RGainControl.DOWN);
+			return;
+		}
+
+		// BGain
+		if (splitProperty[1].equals(Command.BGAIN.getName() + BGainControl.UP.getName())) {
+			bGainControl(BGainControl.UP);
+			return;
+		} else if (splitProperty[1].equals(Command.BGAIN.getName() + BGainControl.DOWN.getName())) {
+			bGainControl(BGainControl.DOWN);
+			return;
+		}
+
+		Command imageProcessCommand = Command.getByName(splitProperty[1]);
+		switch (imageProcessCommand) {
+			case WB_MODE:
+				wbModeControl(WBMode.getByName(value));
+				break;
+
+			case WB_ONE_PUSH_TRIGGER: {
+				wbOnePushTrigger();
+				break;
+			}
+			default: {
+				throw new IllegalStateException("Unexpected value: " + Arrays.toString(splitProperty));
+			}
+		}
+
+	}
+
+	/**
+	 * This method is used to control exposure
+	 */
+	private void exposureControl(String value, Command exposureCommand) {
+		switch (exposureCommand) {
+			case BACKLIGHT: {
+				if (value.equals(SWITCH_STATUS_ON)) {
+					backlightControl(BacklightStatus.ON);
+				} else if (value.equals(SWITCH_STATUS_OFF)) {
+					backlightControl(BacklightStatus.OFF);
+				}
+				break;
+			}
+			case AE_MODE: {
+				aeModeControl(AEMode.getByName(value));
+				break;
+			}
+			case EXP_COMP_DIRECT:
+			case GAIN_LIMIT_DIRECT:
+			case GAIN_DIRECT:
+			case IRIS_DIRECT:
+			case SHUTTER_DIRECT: {
+				float directValue = Float.parseFloat(value);
+				directControl(exposureCommand, (int) directValue);
+				break;
+			}
+			case AUTO_SLOW_SHUTTER: {
+				if (value.equals(SWITCH_STATUS_ON)) {
+					slowShutterControl(SlowShutterStatus.ON);
+				} else if (value.equals(SWITCH_STATUS_OFF)) {
+					slowShutterControl(SlowShutterStatus.OFF);
+				}
+				break;
+			}
+			default: {
+				throw new IllegalStateException("Unexpected value: " + exposureCommand);
+			}
+		}
+	}
+
+	/**
+	 * This method is used to control power
```

Comment:
Same javadoc comment as above. Please check other javadocs as well (the uncommented ones also)

Reply:
yes thank you, I will check of the javadocs of all methods